File: in_Java/0014 Longest Common Prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        String res = "";
        if(strs == null || strs.length == 0) return res;
        int minlen = strs[0].length();
        for(String s : strs){
            if(minlen > s.length()){
                minlen = s.length();
            }
        }
        boolean flag;
        for(int i = 0; i < minlen; i++){
            flag = false;
            char x = strs[0].charAt(i);
            for(int j = 1; j < strs.length; j++){
                if(strs[j].charAt(i) != x){
                    flag = true;
                    break;
                }
            }
            if(!flag){
                res += Character.toString(x);
            }else{
                break;
            }
        }
        return res;
    }
}

File: in_Python/0014 Longest Common Prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        res = ""
        if strs == None or len(strs) == 0:
            return res
        minlen = len(strs[0])
        mins = strs[0]
        for i in range(1, len(strs)):
            if len(strs[i]) < minlen:
                minlen = len(strs[i])
                mins = strs[i]
        for i in range(minlen):
            cur_c = mins[i]
            for sts in strs:
                if sts[i] != cur_c:
                    return res
            res += cur_c
        return res